server/controllers/comment.controller.js

```diff
@@ -126,6 +126,41 @@ class CommentController {
     };
     return Response(res, 201, 'Comment liked successfully', likeObject);
   }
+
+  /**
+    * @description Deletes a user's comment
+    * @param {object} req - The request object
+    * @param {object} res - The response object
+    * @returns {array} An empty array
+    */
+  static async deleteComment(req, res) {
+    const { commentId, slug } = req.params;
+    const { user: { id } } = req;
+    const articleExists = await Article.findOne({
+      where: { slug },
+    });
+    if (!articleExists) return Response(res, 404, 'Article not found.');
+
+    const articleId = articleExists.dataValues.id;
+    const table = articleExists.dataValues.published ? Comment : DraftComment;
+    const commentExists = await table.findOne({
+      where: {
+        id: commentId,
+        userId: id
+      }
+    });
+
+    if (!commentExists) return Response(res, 404, 'Comment not found.');
+
+    await table.destroy({
+      where: { id: commentId }
+    });
+    await table.destroy({
+      where: { parentId: commentId }
+    });
```

Comment:
@TheDrizzyWay good work, but is there any way you can make the table destroy call once?

Reply:
The comment and thread comments are not linked via foreign key, so two queries are needed.

Reply:
@TheDrizzyWay Oh, I see that now. 👍